### ISSUE TITLE
[cloud_firestore ] Firebase snapshot return always 1 document when it is empty

### ISSUE THREAD

**Issue by klaszlo8207** [FirebaseExtended/flutterfire#2161]
See here:

https://stackoverflow.com/questions/60654129/flutter-firebase-snapshot-return-always-1-document-when-it-is-empty#60654129

Is it a BUG? How can I listen on the same collection that is empty?

**Comment by klaszlo8207** [comment]
I think I found the probelem

when I used this query

**_workouts.where("ownerUid", isEqualTo: myUserId).orderBy("order").snapshots();**

then when I had EMPTY items the listener not refreshed

after changed to:

**_workouts.where("ownerUid", isEqualTo: myUserId).snapshots();** 
    
(removed the order)

VOILA, working!

Is it a BUG?

**Comment by iapicca** [comment]
Hi @klaszlo8207 
This platform is not meant for assistance on personal code.
I believe asking on SO was the correct approach,
you may also get some on https://flutter.dev/community
for questions like this

Closing, as this isn't an issue with Flutter itself,
if you disagree please write in the comments and I will reopen it
Thank you

**Comment by klaszlo8207** [comment]
It was a bug in the firestore. Thats why I created here... I think that is not normal if I create a query with orderBy in the end and the listener stop working...

**Comment by iapicca** [comment]
Hi again @klaszlo8207 
can you please provide your `flutter doctor -v` , 
your `flutter run --verbose` (trying receiving the from cf with `.orderBy()`), 
your `pubspec.yaml` ,
and if possible a minimal code sample
of your implementation of `cloud_firestore`
Thank you

**Comment by klaszlo8207** [comment]
Sorry I have deadlines, but the cide looks like this:

Stateful class

Streambuilder in the build

The stream is a collection with a where and an orderBy query

After you add a delete button. When you have only one item left, the delete button deleting the last document from the collection but the listener stop working. 
Always get back 1 item instead of 0.

If you remove the orderBy from the stream end, then it is working.

**Comment by klaszlo8207** [comment]
My solution was to remove to orderBy and just sort the array from dart.

**Comment by Ehesp** [comment]
Leaving this one open to track, maybe `orderBy` is broken. Tests will pick this one up.

**Comment by baku-apps** [comment]
I have similar issue. When using `orderby` property and the collection is empty it fetches the cached item. 

code sample:
```
Firestore.instance
        .collection(FirestoreCollections.ride)
        .where('xxx', isEqualTo: xxx)
        .orderBy('xxx', descending: true)
        .snapshots()
```


Not sure if this is related, but i also get the following exception when using `oderby` on empty collection `Status{code=FAILED_PRECONDITION, description=The query requires an index.`

**Comment by Salakar** [comment]
Hey all, as part of our on-going work for #2582, this should have been resolved in our Firebase Firestore rework (#2913) - which has now been merged into master. We'll look at publishing some prereleases in the next few days. Thank you
